recbole/data/dataset/dataset.py

```diff
@@ -1508,7 +1508,10 @@ def build(self):
     def save(self):
         """Saving this :class:`Dataset` object to :attr:`config['checkpoint_dir']`.
         """
-        file = os.path.join(self.config['checkpoint_dir'], f'{self.config["dataset"]}-dataset.pth')
+        save_dir = self.config['checkpoint_dir']
+        if not os.path.exists(save_dir):
+            os.makedirs(save_dir)
```

Comment:
```suggestion
        ensure_dir(self.config['checkpoint_dir'])
```
We have alread implemented `ensure_dir` function in `recbole.utils`. You can use this function to check directory existance and create it if it does not exist.